numpy/core/src/umath/ufunc_object.c

```diff
@@ -3766,8 +3776,17 @@ PyUFunc_GenericReduction(PyUFuncObject *ufunc, PyObject *args,
     /* Try to interpret axis as an integer */
     else {
         int axis = PyArray_PyIntAsInt(axes_in);
+        int axis_orig = axis;
+        PyObject *error = PyErr_Occurred();
         /* TODO: PyNumber_Index would be good to use here */
-        if (axis == -1 && PyErr_Occurred()) {
+        if (axis == -1 && error != NULL) {
+            /*
+             * Need to re-raise the exact same exception returned by
+             * PyArray_PyIntAsInt to not break the deprecation tests
```

Comment:
I ask, because I'd rather avoid temporary complications in the code like this that may never get resimplified.